### PR TITLE
Improve CI and include coverage data

### PR DESCRIPTION
**Pull request by nelimee**
Replaces #295. The only differences are:
- this PR is from a branch within the `QCHacker` group (and not on the `nelimee/tqec` fork),
- the last 3 commits are removed.

Adding a message to PRs require `write` permission to that PR, which is only given when the branch to be merged is from the same repository as the base branch (here, `QCHackers/tqec`). Trying to circumvent that with `workflow_run` seems too complex for such a low-priority issue, so let's live with that restriction for the moment.

TLDR: PRs from branches in any repository other than `QCHackers/tqec` will fail the CI job about coverage, this is "expected" and is not a code issue but rather a GitHub permission issue.

### CODE REVIEW

**Review by inmzhang — APPROVED** [review]
This looks great!

**Comment by KabirDubey** [comment]
> TLDR: PRs from branches in any repository other than `QCHackers/tqec` will fail the CI job about coverage, this is "expected" and is not a code issue but rather a GitHub permission issue.

Does this mean that from now on, contributors on should work on branches from the `QCHackers` group instead of their own forks? If so, may implicate step 4 of the [contributions file](https://github.com/QCHackers/tqec/blob/main/CONTRIBUTING.md)
